### DISCUSSION TITLE
Conditional backwards link definition on schema

### DISCUSSION THREAD

**Discussion by kafein** [geldata/gel#2700] [Q&A] [answered]
Hello!

I have the following schema.

```
    type Entry {
        multi link revisions := .<entry[IS EntryRevision];
        property revision_count := count(.revisions)

        link active_revision := # I don't know how to make this first EntryRevision which fulfils the condition "activeRevision = true"
    }

    type EntryRevision {
         required property content -> str {
            constraint min_len_value(3);
            constraint max_len_value(10000);
        };

        link entry -> Entry;
      
        property activeRevision -> bool;
    }
```

I want to define a computed link property which fulfils the condition. I think this is possible with the querying but I wonder if it's possible inside the schema definition.

Thanks!

**Comment by msullivan** [comment, marked as answer]
`link active_revision := (SELECT .<entry[IS EntryRevision] FILTER .activeRevision LIMIT 1);` works

`link active_revision := (SELECT .entries FILTER .activeRevision LIMIT 1)` *ought* to work, but I get 
"error: object type 'default::EntryRevision' has no link or property 'activeRevision'", which is a bug I'll look into.
